### ISSUE TITLE
Support self service registration

### ISSUE THREAD

**Issue by kaiyou** [Mailu/Mailu#340]
Currently Mailu only supports account creation by a domain manager. It would be interesting to add two new features:
- account self-service registration
- domain self-service registration

Regarding account registration, an additional field in domains, that can be set by a global administrator, will allow for registration on the domain. Registration will still honor user count limits if set.

Domain registration will first require that the domain owner add a TXT entry to the zone including the Mailu instance main hostname, thus confirming the requester is legitimate for registering the domain. Then an initial account will be created, that is domain manager for the domain.

**Comment by kaiyou** [comment]
A first version of self-service user signup is enabled on master, without any captcha for now. User signup can be enabled for a domain in the domain config.

**Comment by kaiyou** [comment]
This lib was suggested by SaiGonSeamus: https://github.com/lepture/captcha

**Comment by kaiyou** [comment]
Also, I made sure that chosing the domain is a first step process and the domain is specified in the query string, so that one can link to the registration form for a specific domain.

**Comment by kaiyou** [comment]
We'll also have to be a bit more careful when checking the localpart format before releasing this

**Comment by kaiyou** [comment]
Unffortunately, I did not find any proper captcha library for Python flask that can use a Redis backend for storing captchas.

**Comment by whitef0x0** [comment]
@kaiyou is this on hold or permanently closed?

**Comment by kaiyou** [comment]
It is actually implemented and missing a captcha before we can merge it.

**Comment by kaiyou** [comment]
So it should be available in the next release

**Comment by kaiyou** [comment]
After some long hours trying to get something decent working, I finally decided to go with an optional reCaptcha. I added a warning about the obvious privacy issues.

**Comment by kaiyou** [comment]
This is now implemented in master.
